Modules/IO/GDCM/src/itkGDCMImageIO.cxx

```diff
@@ -475,25 +491,32 @@ GDCMImageIO::InternalReadImageInformation()
   m_NumberOfComponents = pixeltype.GetSamplesPerPixel();
   if (image.GetPhotometricInterpretation() == gdcm::PhotometricInterpretation::PALETTE_COLOR)
   {
+    // PALETTE_COLOR is always expanded in RGB image
     itkAssertInDebugAndIgnoreInReleaseMacro(m_NumberOfComponents == 1);
-    // TODO: need to do the LUT ourself...
-    // itkExceptionMacro(<< "PALETTE_COLOR is not implemented yet");
-    // AFAIK ITK user don't care about the palette so always apply it and fake a
-    // RGB image for them
     m_NumberOfComponents = 3;
   }
   if (m_NumberOfComponents == 1)
   {
     this->SetPixelType(SCALAR);
   }
+  else if (m_NumberOfComponents == 3)
+  {
+    const gdcm::PhotometricInterpretation & pi = image.GetPhotometricInterpretation();
+    const bool ybr =
+      pi == gdcm::PhotometricInterpretation::YBR_FULL ||
+      pi == gdcm::PhotometricInterpretation::YBR_FULL_422 ||
+      pi == gdcm::PhotometricInterpretation::YBR_PARTIAL_422 ||
+      pi == gdcm::PhotometricInterpretation::YBR_PARTIAL_420;
+    if ((!m_ReadYBRtoRGB && ybr) || pi == gdcm::PhotometricInterpretation::HSV)
+      this->SetPixelType(VECTOR);
```

Comment:
ITK's style calls for braces `{}` for each `if`, even when it contains only one statement like here.

Reply:
Done